java/jdbc/statement-interceptor/demo-app/src/main/java/com/oracle/jdbc/samples/statementinterceptordemo/dao/impl/EmployeeDAOImpl.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates.
+ *
+ *   This software is dual-licensed to you under the Universal Permissive License
+ *   (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License
+ *   2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose
+ *   either license.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ *
+ */
+
+package com.oracle.jdbc.samples.statementinterceptordemo.dao.impl;
+
+import com.oracle.jdbc.samples.statementinterceptordemo.dao.EmployeeDAO;
+import com.oracle.jdbc.samples.statementinterceptordemo.models.Employee;
+import lombok.extern.java.Log;
+import org.springframework.context.annotation.Primary;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Random;
+import java.util.logging.Level;
+import java.util.stream.IntStream;
+
+@Repository
+@Primary
+@Log
+public class EmployeeDAOImpl implements EmployeeDAO {
+
+  private final JdbcTemplate jdbcTemplate;
+
+  private static final int employeeCount = 5;
+
+  public EmployeeDAOImpl(JdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  @Override
+  public List<Employee> findAll() {
+    final var sql = "SELECT * FROM employees WHERE visible = 1";
+    return jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(Employee.class));
+  }
+
+  @Override
+  public List<Employee> findByName(final String fullName) {
+    // SQL injection vulnerability
+    final var sql =
+      "SELECT * FROM employees WHERE visible = 1 AND full_name='" + fullName
+      + "'";
+    final List<Employee> employees =
+      jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(Employee.class));
+
+    return List.copyOf(employees);
+  }
+
+  @Override
+  public void initializeEmployeesData() {
+    try {
+      jdbcTemplate.execute("""
+                             CREATE TABLE employees (
+                               id CHAR(32) DEFAULT SYS_GUID() PRIMARY KEY,
```

Comment:
NUMBER GENERATED ALWAYS AS IDENTITY

Reply:
Look at this:
https://github.com/oracle-mytodolist/springboot/blob/spring-appstack/src/main/java/com/oracle/todoapp/mytodolist/ToDoItem.java